hathor/conf/settings.py

```diff
@@ -14,17 +14,20 @@
 
 import os
 from math import log
-from typing import List, NamedTuple, Optional
+from typing import Dict, List, Optional, Union
+
+from pydantic import validator
 
 from hathor.checkpoint import Checkpoint
+from hathor.utils import pydantic
 
 DECIMAL_PLACES = 2
 
 GENESIS_TOKEN_UNITS = 1 * (10**9)  # 1B
 GENESIS_TOKENS = GENESIS_TOKEN_UNITS * (10**DECIMAL_PLACES)  # 100B
 
 
-class HathorSettings(NamedTuple):
+class HathorSettings(pydantic.BaseModel):
```

Comment:
What if the yaml file does have a required attribute? What if it has a typo that leads to an unknown attribute?

Reply:
Changed as explained in https://github.com/HathorNetwork/hathor-core/pull/593#issuecomment-1546102674

Reply:
Great! Thanks! Should we add a test with an invalid yaml to confirm the validation is raising an exception?

Reply:
Indeed, I'll add it

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/593/commits/48715e3d8a37dcb59ff1cbb2c3ce848906b8c42a

Reply:
Thanks!